File: src/Net/Store/Before/Salesman.cs
namespace Before
{
    public class Salesman : Employee
    {
        public int CommPor{ get; set; } //CommissionPorcentage

        public decimal MonQuo{ get; private set; } //MonthQuota

        public Salesman(string firstName, string lastName, decimal fixedSalary, int commPor)
            : base(firstName, lastName, fixedSalary)
        {
            this.CommPor = commPor;
        }

        public decimal CalculateNetSalary()
        {
            decimal addicionalBenefits = this.MonQuo * this.CommPor / 100;
            decimal pensionFounds = this.FixedSalary * 10 / 100;
            decimal tax = 0;
            if (FixedSalary > 3500)
                tax = FixedSalary * 5 / 100;
            return addicionalBenefits + FixedSalary - pensionFounds - tax;
        }

        /// <summary>
        /// Update the Month Quota
        /// </summary>
        /// <param name="addQuo">
        /// The addicional quota.
        /// </param>
        public void UpdateMonQuo(decimal addQuo)
        {
            MonQuo = MonQuo + addQuo;
        }
    }
}

File: src/Java/1. First Course/4. Third Hole/src/main/refactoringgolf/store/Salesman.java
package refactoringgolf.store;

public class Salesman extends Employee {

	private int commPor; // CommissionPorcentage
	private float monQuo; // MonthQuota

	public Salesman(String firstName, String lastName, float fixedSalary, int commPor){
    	super(firstName, lastName, fixedSalary);
        this.commPor = commPor;
    }

	public void setCommPor(int commPor) {
		this.commPor = commPor;
	}

	public int getCommPor() {
		return commPor;
	}

	public float getMonQuo() {
		return monQuo;
	}

	public float netSalary() {
		float addicionalBenefits = monQuo * commPor / 100;
		float pensionFounds = fixedSalary * 10 / 100;
		float tax = 0;
		if (fixedSalary > 3500)
			tax = fixedSalary * 5 / 100;
		return addicionalBenefits + fixedSalary - pensionFounds - tax;
	}

	/**
	 * @param addQuo
	 *            The addicional quota
	 */
	public void updateMonQuo(float addQuo) {
		monQuo = monQuo + addQuo;
	}
}
